include/lbann/layers/transform/reshape.hpp

```diff
@@ -99,15 +89,15 @@ class reshape_layer : public transform_layer {
   }
 
   void fp_compute() override {
-    El::LockedView(*this->m_activations_v, *this->m_prev_activations_v);
+    El::LockedView(get_activations(), get_prev_activations());
   }
 
   void bp_compute() override {
-    El::LockedView(*this->m_error_signal_v, *this->m_prev_error_signal_v);
+    El::Axpy(DataType(1), get_prev_error_signals(), get_error_signals());
```

Comment:
I have seen in several places where you use an Axpy instead of a copy.  Is the Axpy parallelized better than the copy?

Reply:
Copying overwrites the error signal, disrupting possible contributions from activation regularization. This introduces a little extra overhead, but it would take some very ugly code to optimize it.

Alternatively, implementing #193 will make sure that the objective function only interacts with the error signal in an objective_function_layer. In this case, overwriting the error signal will be correct and we can perform copies and views instead of axpys.